packages/material-ui-lab/src/Autocomplete/Autocomplete.test.js

```diff
@@ -230,4 +230,140 @@ describe('<Autocomplete />', () => {
+      expect(textbox).to.be.focused;
+      expect(textbox).to.have.attribute('aria-activedescendant', options[0].getAttribute('id'));
+    });
+
+    describe('prop: inlcudeInputInList', () => {
+      it('considers the textbox the predessor of the first option when pressing Up', () => {
+        const { getByRole } = render(
+          <Autocomplete
+            includeInputInList
+            options={['one', 'two', 'three']}
+            renderInput={params => <TextField autoFocus {...params} />}
+          />,
+        );
+        fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' });
+
+        fireEvent.keyDown(document.activeElement, { key: 'ArrowUp' });
+
+        const textbox = getByRole('textbox');
+        expect(textbox).to.be.focused;
+        expect(textbox).not.to.have.attribute('aria-activedescendant');
+      });
+
+      it('considers the textbox the successor of the last option when pressing Down', () => {
+        const { getByRole } = render(
+          <Autocomplete
+            includeInputInList
+            options={['one', 'two', 'three']}
+            renderInput={params => <TextField autoFocus {...params} />}
+          />,
+        );
+        fireEvent.keyDown(document.activeElement, { key: 'ArrowUp' });
+
+        fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' });
+
+        const textbox = getByRole('textbox');
+        expect(textbox).to.be.focused;
+        expect(textbox).not.to.have.attribute('aria-activedescendant');
+      });
+    });
+
+    describe('prop: disableListWrap', () => {
+      it('keeps focus on the first item if focus is on the first item and pressing Up', () => {
+        const { getAllByRole, getByRole } = render(
+          <Autocomplete
+            disableListWrap
+            options={['one', 'two', 'three']}
+            renderInput={params => <TextField autoFocus {...params} />}
+          />,
+        );
+        fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' });
+
+        fireEvent.keyDown(document.activeElement, { key: 'ArrowUp' });
+
+        const textbox = getByRole('textbox');
+        expect(textbox).to.be.focused;
+        expect(textbox).to.have.attribute(
+          'aria-activedescendant',
+          getAllByRole('option')[0].getAttribute('id'),
+        );
+      });
+
+      it('focuses the first item when pressing Up when no option is active', () => {
+        const { getAllByRole, getByRole } = render(
+          <Autocomplete
+            disableListWrap
+            options={['one', 'two', 'three']}
+            renderInput={params => <TextField autoFocus {...params} />}
+          />,
+        );
+
+        fireEvent.keyDown(document.activeElement, { key: 'ArrowUp' });
+
+        const textbox = getByRole('textbox');
+        const options = getAllByRole('option');
+        expect(textbox).to.be.focused;
+        expect(textbox).to.have.attribute('aria-activedescendant', options[0].getAttribute('id'));
```

Comment:
I'm surprised by this one. I would expect that we focus the last option:

> Up Arrow (Optional): If the popup is available, places focus on the last focusable element in the popup.

I could change the behavior with:
```diff
diff --git a/packages/material-ui-lab/src/useAutocomplete/useAutocomplete.js b/packages/material-ui-lab/src/useAutocomplete/useAutocomplete.js
index 07a5f430f..07c3bde56 100644
--- a/packages/material-ui-lab/src/useAutocomplete/useAutocomplete.js
+++ b/packages/material-ui-lab/src/useAutocomplete/useAutocomplete.js
@@ -281,7 +281,7 @@ export default function useAutocomplete(props) {
           return -1;
         }

-        if (disableListWrap || Math.abs(diff) > 1) {
+        if ((disableListWrap && highlightedIndexRef.current !== -1) || Math.abs(diff) > 1) {
           return 0;
         }
```

Reply:
I would've expected this to work different as well. I'll apply your patch and see if this worked.

Reply:
Works :+1: